src/steps/05-PerformanceRequirements/DOW/ComputeForm.spec.ts

```diff
@@ -0,0 +1,700 @@
+      computeForm.vm.$data.errorBag = {810:true, 827: false}
+
+      // change the data.errorBagValues value to activate data reactivity
+      await wrapper.setData({
+        errorBagValues: [true, false]
+      })
+      expect(await wrapper.vm.$data.formHasErrors).toBe(true);
+    });
+  
+    it("validate() - ensures validate() has been called", async ()=>{
+      const validateSpy = await jest.spyOn(wrapper.vm, "validate");
+      wrapper.vm.validate();
+      expect(validateSpy).toHaveBeenCalled();
+    })
+
+    it("createPeriodCheckboxItems() - tests that unsorted SNOW data is successfully transformed to " +
+       "expected sorted datasource array for period checkbox items", async () => {
+      const spy = await jest.spyOn(wrapper.vm, "createPeriodCheckboxItems");
+      const _createPeriodCheckboxItems = wrapper.vm.createPeriodCheckboxItems(
+        [ 
+          {
+            "period_unit": "YEAR",
+            "period_unit_count": "1",
+            "period_type": "BASE",
+            "option_order": "2",
+            "sys_id": "period_02"
+          },
+          {
+            "period_unit": "YEAR",
+            "period_unit_count": "1",
+            "period_type": "BASE",
+            "option_order": "1",
+            "sys_id": "period_01"
+
+          },
+        ]
+      );        
+      expect(spy).toHaveBeenCalled();
+      expect(_createPeriodCheckboxItems).toEqual([
+        {
+          "id": "BASE",
+          "label": "Base period",
+          "value": "period_01"
+        },
+        {
+          "id": "OPTION1",
+          "label": "Option period 1",
+          "value": "period_02"
+        }
+      ])
+    });
+    it("createCheckboxOrRadioItems() - tests that SNOW data is successfully transformed to " +
+        "expected datasource array for checkbox or radio items", async()=>{
+      const spy = await jest.spyOn(wrapper.vm, "createCheckboxOrRadioItems");
+      // creates array to be used for Checkbox/RadioList datasource
+      const _createCheckboxOrRadioItems = wrapper.vm.createCheckboxOrRadioItems(
+        allClassificationLevels, 
+        "Radio"
+      );
+      expect(spy).toHaveBeenCalled();
+      // ensures one Checkbox/RadioList object in datasource is crafted correctly
+      expect(_createCheckboxOrRadioItems.filter((rdoItem: Checkbox)=>{
+        return rdoItem.value === 'class1' && rdoItem.label === 'Unclassified / Impact Level 4 (IL4)' 
+      })).toHaveLength(1);
+    });
+  })
+
+ 
+  describe("testing modal functionality...", () => {
+    it("if model opens, set data.showDialog=true", async () => {
+      expect(wrapper.vm.$data.showDialog).toBe(false);
+      const spy = await jest.spyOn(wrapper.vm, "openModal");
+      await wrapper.vm.openModal();
+      expect(spy).toHaveBeenCalled();
+      expect(wrapper.vm.$data.showDialog).toBe(true);
+    });
+    it("if modal closed, expect data.showDialog=false", async () => {
+      expect(wrapper.vm.$data.showDialog).toBe(false);
```

Comment:
We don't have any `//given` or `//expects` behaviors, so what are we really testing here?

Reply:
Page load, the data.showDialog should be set to false.